File: 561. Array Partition/561.go
// 561. Array Partition solution in GoLang
// https://leetcode.com/problems/array-partition/description/

package main

import "sort"

func arrayPairSum(nums []int) int {
	sort.Ints(nums)

	max_sum := 0

	for i := 0; i < len(nums); i += 2 {
		max_sum += nums[i]
	}

	return max_sum
}

File: 561. Array Partition/561.js
// 561. Array Partition solution in JavaScript
// https://leetcode.com/problems/array-partition/description/

/**
 * @param {number[]} nums
 * @return {number}
 */
 var arrayPairSum = function(nums) {
    nums.sort((a, b) => a - b);

    let max_sum = 0;

    for (let i = 0; i < nums.length; i += 2) {
        max_sum += nums[i];
    }

    return max_sum;
};